MapboxNavigation/InstructionPresenter.swift

```diff
@@ -244,9 +244,14 @@ class InstructionPresenter {
     }
     
     private func takeSnapshot(on view: UIView) -> UIImage? {
-        let window = UIApplication.shared.delegate!.window!!
+        let window: UIWindow
+        if let hostView = dataSource as? UIView, let hostWindow = hostView.window {
+            window = hostWindow
+        } else {
+            window = UIApplication.shared.delegate!.window!!
```

Comment:
Should this be more gentle?

Reply:
We could return nil here, but isn't it a fair assumption make that the application will always have a window on the phone? 

if we ever end up breaking that assumption, it might be a better idea to fail loudly.

Reply:
This line is unchanged in this PR compared to master. This PR only provides a gentler alternative when rendering for CarPlay.